src/sentry/api/serializers/models/group.py

```diff
@@ -576,8 +576,16 @@ def _get_seen_stats(self, item_list, user):
                 group_ids,
                 self.environment_ids,
             )
+
+            first_seen_data = {
+                ge['group_id']: ge['first_seen__min'] for ge in GroupEnvironment.objects.filter(
+                    group_id__in=[item.id for item in item_list],
+                    environment_id__in=self.environment_ids,
+                ).values('group_id').annotate(Min('first_seen'))
+            }
+
             for item_id, value in seen_data.items():
-                first_seen[item_id] = value['first_seen']
+                first_seen[item_id] = first_seen_data.get(item.id, value['first_seen'])
```

Comment:
Since this is backfilled, is there still a case where we might not be able to get the `first_seen` value from `GroupEnvironment`?

Reply:
There are some GroupEnvironments that still have a null value for first_seen -- ~5%. From spot checking these, it seemed like they were orphaned and that either the group or environment (or both) had been deleted. You're probably right though, that it'd maybe be better for now to not fall back to snuba so that it's more obvious if the data is missing when it doesn't seem like it should be.

Reply:
Right, I forgot that the backfill couldn't completely fill that field